examples/node-regression-with-subgraph-and-graph-sample.ipynb

```diff
@@ -0,0 +1,713 @@
+    "    MATCH (c:Crocodile {id: row.id})\n",
+    "    SET c.target = toInteger(row.target)\n",
+    "\"\"\",\n",
+    "    {\"baseUrl\": baseUrl},\n",
+    ")\n",
+    "gds.run_cypher(\n",
+    "    \"\"\"\n",
+    "    LOAD CSV WITH HEADERS FROM $baseUrl + '/squirrel/musae_squirrel_target.csv' AS row\n",
+    "    MATCH (s:Squirrel {id: row.id})\n",
+    "    SET s.target = toInteger(row.target)\n",
+    "\"\"\",\n",
+    "    {\"baseUrl\": baseUrl},\n",
+    ")"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 514,
+   "outputs": [
+    {
+     "data": {
+      "text/plain": "Empty DataFrame\nColumns: []\nIndex: []",
+      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n    </tr>\n  </thead>\n  <tbody>\n  </tbody>\n</table>\n</div>"
+     },
+     "execution_count": 514,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Create feature vectors\n",
+    "gds.run_cypher(\n",
+    "    \"\"\"\n",
+    "    CALL apoc.load.json($baseUrl + '/chameleon/musae_chameleon_features.json') YIELD value\n",
+    "    WITH value, keys(value) AS keys\n",
+    "    UNWIND keys AS key\n",
+    "    WITH value[key] AS feature, key\n",
+    "    MATCH (c:Chameleon {id: key})\n",
+    "    SET c.features = feature\n",
+    "\"\"\",\n",
+    "    {\"baseUrl\": baseUrl},\n",
+    ")\n",
+    "gds.run_cypher(\n",
+    "    \"\"\"\n",
+    "    CALL apoc.load.json($baseUrl + '/crocodile/musae_crocodile_features.json') YIELD value\n",
+    "    WITH value, keys(value) AS keys\n",
+    "    UNWIND keys AS key\n",
+    "    WITH value[key] AS feature, key\n",
+    "    MATCH (c:Crocodile {id: key})\n",
+    "    SET c.features = feature\n",
+    "\"\"\",\n",
+    "    {\"baseUrl\": baseUrl},\n",
+    ")\n",
+    "gds.run_cypher(\n",
+    "    \"\"\"\n",
+    "    CALL apoc.load.json($baseUrl + '/squirrel/musae_squirrel_features.json') YIELD value\n",
+    "    WITH value, keys(value) AS keys\n",
+    "    UNWIND keys AS key\n",
+    "    WITH value[key] AS feature, key\n",
+    "    MATCH (c:Squirrel {id: key})\n",
+    "    SET c.features = feature\n",
+    "\"\"\",\n",
+    "    {\"baseUrl\": baseUrl},\n",
+    ")"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "# Preparing the dataset for the pipeline\n",
+    "\n",
+    "In order to use the dataset, we must make sure that the feature dimension is consistent over the entire node sets.\n",
+    "In their raw form, the feature vectors are of different lengths.\n",
+    "To overcome this, we will use a one-hot encoding.\n",
```

Comment:
i would say the main reason to use one-hot encoding is that we dont support categorical id's or sets of those as features. i dont think other libraries do either because its not something a typical model understands and you could perhaps say they are not even features. 

in a way you could say a side effect of the one-hot encoding is that you get same length feature vectors.

Maybe you could write that the data is not in an ml-friendly format because most or all ml-models cannot work with features whose values are id's. using one-hot encoding transforms the data to ml-friendly vectors (?)

Reply:
Solved together!